### PR TITLE
docker optimization - copy local slips files, pip3 using requirements…

### PR DESCRIPTION
**Pull request by HappyStoic**
Some basic changes in `docker/ubuntu-image/Dockerfile`.

**Changes:**
* Copy local Slips version into the image (instead of `git clone`)
* Install Python requirements from `requirements.txt` file (instead of naming all the dependencies in the Dockerfile beacuse it creates duplication of dependencies in multiple locations which is hard to keep track of and hard to update)
    - If later conda (or other python environment managment tool shall be used, this can be changed into one Dockerfile stage (stages explained below)
* Simple Makefile to build an image using `make image` and exec shall in the image using `make bash`
* Update readme with proper `docker build ...` command 

**To check:**
* Are dependencies in `requirements.txt` correct? I moved dependencies from Dockerfile into the requirements.txt file but there were not not provided versions in the Dockerfile so I tried to make an educated guess or leave some without a version :smile: 

**More future improvement:**
* Play with an order of the steps so docker cache is most utilized
* Create `package.json` and `package.lock` for node `kalipso` module dependencies (I believe these files are created in `p2p` module pull request) and then install dependencies during build using this file (now `kalipso` dependencies are duplicated in `Dockerfile` and `install.sh` script. It shall be just one place and preferably the node's package.json)  
* Use Dockerfile build stages to compile some dependencies. Perfect example for dockerfile stages would be compilation of `pigeon` for p2p module during the build. I will do it later

*Tldr; Dockerfile stage is simply another separated environment during the build which can be used for anything and later final image can copy files from these stages to itself. So for example one stage can be docker image with `golang` compilator used just to compile `pigeon`. Later you copy compiled pigeon binary into the final image without having useless `golang` compilator in final image.*

### CODE REVIEW

**Comment by AlyaGomaa** [comment]
Copied your changed to another branch and merged to master and develop, your branch has merge conflicts because it was branched off of master instead of develop
